scripts/docs_config/introduction.rst

```diff
@@ -0,0 +1,101 @@
+for memory management. UMF allows users to manage multiple memory pools 
+characterized by different attributes, allowing certain allocation types to be 
+isolated from others and allocated using different hardware resources as 
+required. 
+
+A memory pool is a combination of a pool allocator and one or more memory 
+targets accessed by memory providers along with their properties and allocation 
+policies. Specifically, a memory provider is responsible for coarse-grained 
+memory allocations, while the pool allocator controls the pool and handles 
+fine-grained memory allocations. UMF provides distinct interfaces for both pool 
+allocators and memory providers, allowing integration into various 
+applications. 
+
+.. figure:: ../assets/images/intro_architecture.png
+
+The UMF library contains various pool allocators and memory providers  but also 
+allows for the integration of external ones, giving users the flexibility to 
+either use existing solutions or provide their implementations. 
+
+Memory Providers
+================
+
+A memory provider is an abstraction for coarse (memory page) allocations and 
+deallocations of target memory types, such as host CPU, GPU, or CXL memory. 
+A single memory provider kind can efficiently manage the memory operations for 
+one or multiple devices within the system or other memory sources like 
+file-backed or user-provided memory.
+
+UMF comes with several bundled memory providers. Please refer to the README.md 
+to see a full list of them. There is also a possibility to use externally 
+defined memory providers if they implement the UMF interface.
+
+To instantiate a memory provider, user must pass an additional context with 
+contains the details about the specific memory target that should be used. This 
+would be a NUMA node mask for the OS memory provider, file path for the 
+file-backed memory provider, etc. After creation, the memory provider context
+can't be changed.
+
+Pool Allocators
+===============
+
+A pool allocator is an abstraction over object-level memory management based 
+on coarse chunks acquired from the memory provider. It manages the memory pool 
+and services fine-grained malloc/free requests. 
+
+Pool allocators can be implemented for be general purpose or to fulfill 
+specific use cases. Implementations of the pool allocator interface can 
+leverage existing allocators (e.g., jemalloc or oneTBB) or be fully 
+customizable. The pool allocator abstraction could contain basic memory 
+management interfaces, as well as more complex ones that can be used, for 
+example, by the implementation for page monitoring or control (e.g., `madvise`).
+
+UMF comes with several bundled memory providers. Please refer to the README.md 
+to see a full list of them. There is also a possibility to use externally 
+defined pool allocators if they implement the UMF interface.
+
+Memory Pools
+============
+
+A Memory pool is a combination of a pool allocator and one or more memory 
```

Comment:
`Memory pool consists of a pool allocator and a single or multiple memory targets ...`

Reply:
done